diff -up linbox-1.3.2/interfaces/driver/lb-blackbox-functor.h.orig linbox-1.3.2/interfaces/driver/lb-blackbox-functor.h
--- linbox-1.3.2/interfaces/driver/lb-blackbox-functor.h.orig	2012-07-03 17:48:30.630565374 -0400
+++ linbox-1.3.2/interfaces/driver/lb-blackbox-functor.h	2012-07-03 17:48:49.880566113 -0400
@@ -53,7 +53,7 @@ public:
  * Macro for automatic code generation *
  ***************************************/
 #define LB_BLACKBOX_VISIT(B)				\
-void visit(const B &d){apply(d);}
+void visit(const B &d){this->apply(d);}
 
 
 /**********************************************************************************
diff -up linbox-1.3.2/interfaces/driver/lb-domain-functor.h.orig linbox-1.3.2/interfaces/driver/lb-domain-functor.h
--- linbox-1.3.2/interfaces/driver/lb-domain-functor.h.orig	2012-07-03 17:49:05.931566774 -0400
+++ linbox-1.3.2/interfaces/driver/lb-domain-functor.h	2012-07-03 17:49:17.856567170 -0400
@@ -52,7 +52,7 @@ public:
  * Macro for automatic code generation *
  ***************************************/
 #define LB_DOMAIN_VISIT(D)				\
-void visit(const DomainEnvelope<D> &d){apply(d);}
+void visit(const DomainEnvelope<D> &d){this->apply(d);}
 
 
 /*******************************************************************************
diff -up linbox-1.3.2/interfaces/driver/lb-vector-functor.h.orig linbox-1.3.2/interfaces/driver/lb-vector-functor.h
--- linbox-1.3.2/interfaces/driver/lb-vector-functor.h.orig	2012-07-03 17:49:30.130567691 -0400
+++ linbox-1.3.2/interfaces/driver/lb-vector-functor.h	2012-07-03 17:49:42.232568167 -0400
@@ -54,7 +54,7 @@ public:
  * Macro for automatic code generation *
  ***************************************/
 #define LB_VECTOR_VISIT(B)				\
-	void visit(const B &d){apply(d);}
+	void visit(const B &d){this->apply(d);}
 
 
 /**********************************************************************************
diff -up linbox-1.3.2/linbox/algorithms/blackbox-container-symmetric.h.orig linbox-1.3.2/linbox/algorithms/blackbox-container-symmetric.h
--- linbox-1.3.2/linbox/algorithms/blackbox-container-symmetric.h.orig	2012-07-03 17:49:56.067568648 -0400
+++ linbox-1.3.2/linbox/algorithms/blackbox-container-symmetric.h	2012-07-03 17:50:24.032569723 -0400
@@ -59,7 +59,7 @@ namespace LinBox
 		template<class Vector>
 		BlackboxContainerSymmetric (const Blackbox *D, const Field &F, const Vector &u0) :
 			BlackboxContainerBase<Field, _Blackbox> (D, F)
-		{ init (u0, u0); }
+		{ this->init (u0, u0); }
 		BlackboxContainerSymmetric (const Blackbox *D, const Field &F, RandIter &g) :
 			BlackboxContainerBase<Field, _Blackbox> (D, F)
 		{ this->init (g); }
