diff -up linbox-1.3.2/linbox/field/Modular/modular-int64.h.orig linbox-1.3.2/linbox/field/Modular/modular-int64.h
--- linbox-1.3.2/linbox/field/Modular/modular-int64.h.orig	2012-07-03 17:34:48.498533895 -0400
+++ linbox-1.3.2/linbox/field/Modular/modular-int64.h	2012-07-03 17:34:51.880534136 -0400
@@ -96,7 +96,7 @@ namespace LinBox
 	public:
 		typedef int64_t Element;
 
-		typedef FFPACK::Modular<int64_t> Father_t
+		typedef FFPACK::Modular<int64_t> Father_t;
 		friend class FieldAXPY<Modular<int64_t> >;
 		friend class DotProductDomain<Modular<int64_t> >;
 		friend class MVProductDomain<Modular<int64_t> >;
